packages/hardhat/contracts/Ethernauts.sol

```diff
@@ -52,6 +52,8 @@ contract Ethernauts is ERC721Enumerable, Ownable {
 
     bool public permanentUrl;
 
+    string public provenanceHash;
```

Comment:
Should this be:
1) immutable and grouped together with the variables maxTokens, maxGiftable, etc.
2) A bytes32. No need for a dynamic string.